### PR TITLE
Add a script for deploy in debug mode

### LINKED ISSUE #478: Create a script for deploying kubearchive in debug mode
With the changes of https://github.com/kubearchive/kubearchive/issues/442 we are just using `helm install` command for deploying KubeArchive in debug mode.

This is not consistent with the rest of the deployment instructions as we are moving out from the `helm install` command so we are not dealing with the helm charts in the cluster anymore and with the `helm uninstall` or `helm upgrade` command.

We should provide new helper scripts to deploy KubeArchive in debug mode that address the following:

* Receive a parameter `operator` `sink` or `api` to debug
* Build that image in debug mode with `ko`
* (Re)Apply the proper kubernetes resources to successfully use that image and expose the debugging ports

### PR DESCRIPTION
**Pull request by maruiz93**
## Which Issue(s) this Pull Request Resolves

Resolves #478

## Release Note

```release-note
NONE
```

## Notes for Reviewers

I end up using the `jq` + `kubectl` apply after trying several approaches with `kubectl patch` but none seemed to fit very good.
Note that we want to remove the probe keys and that is difficult to do with `kubectl patch`.
I also replaced `apply` for `replace` in the `hack/kubearchive-install.sh` because if not, the services keep the `debug` port.

### CODE REVIEW

File: hack/kubearchive-install.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ helm template kubearchive charts/kubearchive -n kubearchive \
     --include-crds \
     --set "global.production=true" > /tmp/kubearchive-not-resolved.yaml
 ko resolve -f /tmp/kubearchive-not-resolved.yaml --base-import-paths > ${YAML}
-kubectl apply -n kubearchive -f ${YAML}
+kubectl replace -n kubearchive -f ${YAML}
```
Comment (rh-hemartin) [inline hack/kubearchive-install.sh:22]: I'm testing the PR locally and this does not work if I run the normal process, I think you changed this after installing for the first time. So this command can't `replace` if there are no resources. The process to debug another component needs to change to `hack/kubearchive-delete.sh` and then `hack/kubearchive-install.sh` again.

Comment (rh-hemartin) [reply hack/kubearchive-install.sh:22]: You can see the error same error I got in the integration tests.

**Comment by rh-hemartin** [comment]
This PR needs to remove the corresponding stuff from the Helm chart.

File: hack/kubearchive-install.sh (hunk 1 above)
Comment (maruiz93) [reply hack/kubearchive-install.sh]: Right, I tested just with some resources and see that even when they were not created with replace they were, but it seems that this doesn't apply for every resource in our script.

I've rollbacked the change in this file

**Comment by maruiz93** [comment]
> This PR needs to remove the corresponding pieces from the Helm chart.

Included

File: test/debug/patch-security-context.yaml
Comment (skoved) [inline test/debug/patch-security-context.yaml:1]: I think you should add `---` to the top of the file to get rid of the linter warning

File: test/debug/patch-service.yaml
Comment (skoved) [inline test/debug/patch-service.yaml:1]: I think you should add --- to the top of the file to get rid of the linter warning

**Review by rh-hemartin — APPROVED** [review]

File: test/debug/patch-security-context.yaml
Comment (maruiz93) [reply test/debug/patch-security-context.yaml:1]: Done!

**Pull request merged**